lib/components/delete_prompts.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:finamp/components/confirmation_prompt_dialog.dart';
+import 'package:finamp/components/global_snackbar.dart';
+import 'package:finamp/models/finamp_models.dart';
+import 'package:finamp/services/downloads_service.dart';
+import 'package:finamp/services/finamp_settings_helper.dart';
+import 'package:finamp/services/jellyfin_api_helper.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+Future<void> askBeforeDeleteDownloadFromDevice(
+    BuildContext context, DownloadStub stub) async {
+  String type = stub.baseItemType.name;
+  await showDialog(
+      context: context,
+      builder: (context) => ConfirmationPromptDialog(
+          promptText: AppLocalizations.of(context)!
+              .deleteFromTargetDialogText("", "device", type),
+          confirmButtonText: AppLocalizations.of(context)!
+              .deleteFromTargetConfirmButton("device"),
+          abortButtonText: AppLocalizations.of(context)!.genericCancel,
+          onConfirmed: () async {
+            try {
+              await GetIt.instance<DownloadsService>().deleteDownload(stub: stub);
+              GlobalSnackbar
+                .message((_) => AppLocalizations.of(context)!
+                      .itemDeletedSnackbar("device", type)
+                );
+
+              if (context.mounted && FinampSettingsHelper.finampSettings.isOffline) {
+                Navigator.of(context).pop();
+              }
+            } catch (err) {
+              GlobalSnackbar.error(err);
+            }
+          },
+          onAborted: () {},
+          centerText: true));
+}
+
+Future<void> askBeforeDeleteDownloadFromServer(
```

Comment:
can we rename this? we don't delete a download from the server. we delete a file/item from the server, and potentially also delete a download from the device :)